linkme-chatapp/devServer.js

```diff
@@ -68,35 +72,42 @@ function createApp() {
     if (!layerProviderID) return res.status(500).send('Couldn\'t find LAYER_PROVIDER_ID');
     if (!layerKeyID) return res.status(500).send('Couldn\'t find LAYER_KEY_ID');
     if (!privateKey) return res.status(500).send('Couldn\'t find Private Key');
-
-    var header = JSON.stringify({
-      typ: 'JWT',           // Expresses a MIMEType of application/JWT
-      alg: 'RS256',         // Expresses the type of algorithm used to sign the token, must be RS256
-      cty: 'layer-eit;v=1', // Express a Content Type of application/layer-eit;v=1
-      kid: layerKeyID
-    });
-
-    var currentTimeInSeconds = Math.round(new Date() / 1000);
-    var expirationTime = currentTimeInSeconds + 10000;
-
-    var claim = JSON.stringify({
-      iss: layerProviderID,       // The Layer Provider ID
-      prn: userId,                // User Identifier
-      iat: currentTimeInSeconds,  // Integer Time of Token Issuance 
-      exp: expirationTime,        // Integer Arbitrary time of Token Expiration
-      nce: nonce                  // Nonce obtained from the Layer Client SDK
-    });
-
-    var jws = null;
-    try {
-      jws = jsrsasign.jws.JWS.sign('RS256', header, claim, privateKey);
-    } catch (e) {
-      return res.status(500).send('Could not create signature. Invalid Private Key: ' + e);
-    }
-
-    res.json({
-      identity_token: jws
-    });
+    
+    validateFacebookUserToken(req.body.user_id, req.body.user_token, function onValidate(successful) {
+      if(!successful) {
+        res.send(401, { error: 'Authentication failed' });
+        return;
+      }
+      
+      var header = JSON.stringify({
+        typ: 'JWT',           // Expresses a MIMEType of application/JWT
+        alg: 'RS256',         // Expresses the type of algorithm used to sign the token, must be RS256
+        cty: 'layer-eit;v=1', // Express a Content Type of application/layer-eit;v=1
+        kid: layerKeyID
+      });
+
+      var currentTimeInSeconds = Math.round(new Date() / 1000);
+      var expirationTime = currentTimeInSeconds + 10000;
+
+      var claim = JSON.stringify({
```

Comment:
to avoid devServer getting out of hand we can put all this in a token provider/manager whatever module 😄